src/overlays/actors/ovl_En_Syateki_Man/z_en_syateki_man.c

```diff
@@ -25,41 +59,376 @@ const ActorInit En_Syateki_Man_InitVars = {
+        this->blinkTimer = decrBlinkTimer;
+    } else {
+        this->blinkFunc = EnSyatekiMan_Blink;
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Syateki_Man/func_80B10EB0.s")
+void EnSyatekiMan_Blink(EnSyatekiMan* this) {
+    s16 decrBlinkTimer = this->blinkTimer - 1;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Syateki_Man/func_80B11164.s")
+    if (decrBlinkTimer != 0) {
+        this->blinkTimer = decrBlinkTimer;
+    } else {
+        s16 nextEyeState = this->eyeState + 1;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Syateki_Man/func_80B111D4.s")
+        if (nextEyeState >= 3) {
+            this->eyeState = 0;
+            this->blinkTimer = 20 + (s32)(Math_Rand_ZeroOne() * 60.0f);
+            this->blinkFunc = EnSyatekiMan_BlinkWait;
+        } else {
+            this->eyeState = nextEyeState;
+            this->blinkTimer = 1;
+        }
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Syateki_Man/func_80B112A0.s")
+void EnSyatekiMan_Update(Actor* thisx, GlobalContext* globalCtx) {
+    s32 pad;
+    EnSyatekiMan* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Syateki_Man/func_80B11310.s")
+    if (this->timer != 0) {
+        this->timer--;
+    }
+    this->actionFunc(this, globalCtx);
+    EnSyatekiMan_Jukebox();
+    this->blinkFunc(this);
+    this->actor.posRot2.pos.y = 70.0f;
+    Actor_SetHeight(&this->actor, 70.0f);
+    func_80038290(globalCtx, &this->actor, &this->headRot, &this->bodyRot, this->actor.posRot2.pos);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Syateki_Man/func_80B11344.s")
+s32 func_80B1148C(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3f* pos, Vec3s* rot, void* thisx) {
+    EnSyatekiMan* this = THIS;
+    s32 turnDirection;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Syateki_Man/EnSyatekiMan_Update.s")
+    if (limbIndex == 1) {
+        rot->x += this->bodyRot.y;
+    }
+    if (limbIndex == 8) {
+        *dList = D_06007E28;
+        turnDirection = 1;
+        if (this->gameResult == RESULT_REFUSE) {
+            turnDirection = -1;
+        }
+        rot->x += this->headRot.y * turnDirection;
+        rot->z += this->headRot.z;
+    }
+    return false;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Syateki_Man/func_80B1148C.s")
+void EnSyatekiMan_Draw(Actor* thisx, GlobalContext* globalCtx) {
+    s32 pad;
+    EnSyatekiMan* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Syateki_Man/EnSyatekiMan_Draw.s")
+    func_80093D18(globalCtx->state.gfxCtx);
+    SkelAnime_DrawFlexOpa(globalCtx, this->skelAnime.skeleton, this->skelAnime.limbDrawTbl, this->skelAnime.dListCount,
+                          func_80B1148C, NULL, this);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Syateki_Man/func_80B1156C.s")
+void EnSyatekiMan_Jukebox(void) {
+    if (BREG(80)) {
```

Comment:
idk if id call the function this, it can be pretty misleading. ChangeBgm seems fine